akka-remote/src/main/scala/akka/remote/serialization/DaemonMsgCreateSerializer.scala

```diff
@@ -149,7 +153,13 @@ private[akka] final class DaemonMsgCreateSerializer(val system: ExtendedActorSys
       val dispatcher =
         if (protoDeploy.hasDispatcher) protoDeploy.getDispatcher
         else NoDispatcherGiven
-      Deploy(protoDeploy.getPath, config, routerConfig, scope, dispatcher)
+
+      val tags: Set[String] =
+        if (protoDeploy.getTagsCount == 0) Set.empty
+        else protoDeploy.getTagsList.iterator().asScala.toSet
+      val deploy = Deploy(protoDeploy.getPath, config, routerConfig, scope, dispatcher)
+      if (tags.isEmpty) deploy
+      else deploy.withTags(tags)
```

Comment:
Or evaluate once vs twice?
```
val isEmpty = protoDeploy.getTagsCount == 0
val tags: Set[String] =
  if (isEmpty) Set.empty
  else protoDeploy.getTagsList.iterator().asScala.toSet
val deploy = Deploy(protoDeploy.getPath, config, routerConfig, scope, dispatcher)
if (isEmpty) deploy
else deploy.withTags(tags)
```

Reply:
Set has a specialized empty impl. so isEmpty is a method that just return `0` so I don't think that matters much from a perf. point of view (dispatch to the method potentially then, but in the context of a copying serializer I don't think that matters much).